moonstone/parsers/counts/taxonomy/metaphlan3.py

```diff
@@ -13,11 +13,52 @@ class Metaphlan3Parser(TaxonomyCountsBase, BaseParser):
     NCBI_tax_column = 'NCBI_tax_id'
 
     def __init__(self, *args, **kwargs):
+        self.analysis_type = kwargs['analysis_type']
+        kwargs.pop('analysis_type')
         super().__init__(*args, parsing_options={'skiprows': 1}, **kwargs)
 
+    def remove_duplicates(self, df):
+        df = df.set_index('clade_name')
+        if self.analysis_type == 'rel_ab':
+            total = 99.9999
+        else:
+            total = df[df.index.map(lambda x: len(x.split('|'))) == 1].sum()
+
+        # dataframe at rank level
+        new_df = df.index.map(lambda x: len(x.split('|')))
+        self._rank_level = new_df.max()
```

Comment:
It does not seem to be reused at the moment so it does not necessary need to be a instance attribute? Or an alternative could be to perform the first steps of this method in the constructor to kind of "initialize" the different parameters and therefore be able to know easily from the parser what is the `rank_level`.

Reply:
I can remove this from the attributes right here. I kept it because later on this instance attribute is initialize so I was 'why not initialize it now?', but as it is re-define anyway later on. I can remove it from here

Reply:
Okay I get it I think! I would suggest to define this attribute elsewhere, maybe in the constructor?

I see that you are using it in the `to_dataframe` method and it is a bit confusing, if you want to find where this `self._rank_level` is defined, it is not straightforward to find.

Reply:
Sorry, I didn't see your reply. I didn't code this part of the code, but self._rank_level was defined in the split_taxa_fill_none method in moonstone/utils/taxonomy.py. I'll see about putting it in the constructor. It does make more sense